runelite-client/src/main/java/net/runelite/client/plugins/driftnet/DriftNetPlugin.java

```diff
@@ -0,0 +1,296 @@
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Inject;
+import lombok.Getter;
+import net.runelite.api.Actor;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.Client;
+import net.runelite.api.GameObject;
+import net.runelite.api.GameState;
+import net.runelite.api.NPC;
+import net.runelite.api.NpcID;
+import net.runelite.api.NullObjectID;
+import net.runelite.api.Player;
+import net.runelite.api.Varbits;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.GameObjectDespawned;
+import net.runelite.api.events.GameObjectSpawned;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.InteractingChanged;
+import net.runelite.api.events.NpcDespawned;
+import net.runelite.api.events.NpcSpawned;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.OverlayManager;
+
+@PluginDescriptor(
+	name = "Drift Net",
+	description = "Display information about drift nets",
+	tags = {"hunter", "fishing", "drift", "net"},
+	enabledByDefault = false
+)
+public class DriftNetPlugin extends Plugin
+{
+	static final String CONFIG_GROUP = "driftnet";
+	private static final int UNDERWATER_REGION = 15008;
+	private static final String CHAT_PRODDING_FISH  = "You prod at the shoal of fish to scare it.";
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private DriftNetConfig config;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private DriftNetOverlay overlay;
+
+	@Getter
+	private Set<NPC> fish = new HashSet<>();
+	@Getter
+	private Map<NPC, Integer> taggedFish = new HashMap<>();
+	@Getter
+	private final List<DriftNet> NETS = ImmutableList.of(
+		new DriftNet(NullObjectID.NULL_31433, Varbits.NORTH_NET_STATUS, Varbits.NORTH_NET_CATCH_COUNT),
+		new DriftNet(NullObjectID.NULL_31434, Varbits.SOUTH_NET_STATUS, Varbits.SOUTH_NET_CATCH_COUNT));
+
+	@Getter
+	private boolean inDriftNetArea;
+	private boolean armInteraction;
+
+	@Provides
+	DriftNetConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(DriftNetConfig.class);
+	}
+
+	@Override
+	protected void startUp()
+	{
+		overlayManager.add(overlay);
+
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			clientThread.invokeLater(() ->
+			{
+				inDriftNetArea = checkArea();
+				updateDriftNetVarbits();
+			});
+		}
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		overlayManager.remove(overlay);
+		reset();
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		switch (event.getGameState())
+		{
+			case LOGIN_SCREEN:
+			case HOPPING:
```

Comment:
Add `LOADING` to this set of cases so that we can ensure fish lists are cleared.
```suggestion
			case HOPPING:
			case LOADING:
```